### ISSUE TITLE
Spek does not output failed assertion messages in Maven

### ISSUE THREAD

**Issue by devonhumes** [JetBrains/spek#303]
## Steps to reproduce
1. Create failing Spek test
2. Run with `mvn -e test`
3. Build fails

### Actual result
The build fails like it should, but the error message causes a NullPointerException. 

```
Running org.devonhumes.SimpleSpec
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.067 sec <<< FAILURE! - in org.devonhumes.SimpleSpec
it runs  Time elapsed: 0.057 sec  <<< FAILURE!
null

Results :

Failed tests: 

```

This is caused by a null pointer exception inside of Surefire when it's trying to print the message.

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project spek-with-maven-sample: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test failed. NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project spek-with-maven-sample: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test failed.
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test failed.
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
        ... 20 more
Caused by: java.lang.NullPointerException
        at org.apache.maven.plugin.surefire.report.DefaultReporterFactory.printTestFailures(DefaultReporterFactory.java:346)
        at org.apache.maven.plugin.surefire.report.DefaultReporterFactory.runCompleted(DefaultReporterFactory.java:137)
        at org.apache.maven.plugin.surefire.report.DefaultReporterFactory.close(DefaultReporterFactory.java:106)
        at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:208)
        at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1026)
        at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:862)
        at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:755)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        ... 21 more
```

See https://github.com/apache/maven-surefire/blob/surefire-2.19.1/maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/report/DefaultReporterFactory.java#L346

### Expected result
It successfully prints the assertion message.

### Demo
https://github.com/devonhumes/spek-kotlin-maven-error-output-null

**Comment by devonhumes** [comment]
Additionally, running these tests in JUnit 5 with the Jupiter engine works just fine.

```
package org.devonhumes

import org.assertj.core.api.Assertions.assertThat;
import org.junit.jupiter.api.Test

class SimpleTest {

    @Test
    fun simpleTest() {
        assertThat(true).isEqualTo(false);
    }
}
```

**Comment by artem-zinnatullin** [comment]
@devonhumes unfortunately none of project maintainers use Maven nowadays… (maybe Hadi, but I'm not sure)

Let's try upgrade surefire plugin to latest version atm: `2.20.1`, here is [link to all releases](https://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22org.apache.maven.plugins%22%20AND%20a%3A%22maven-surefire-plugin%22).

If that doesn't help, then you'll most likely have to open issue against surefire plugin since cause of exception is there as shown by your stacktrace, we'll be happy to follow progress on issue resolution if you post link back!

**Comment by raniejade** [comment]
@artem-zinnatullin JUnit 5 only supports `2.19.1`.

I did some initial digging but I'm lost. It might be good to ask the junit5 team as they are the ones providing the surefire provider.

**Comment by devonhumes** [comment]
Ok I can see that. Unfortunately, I'm stuck with Maven. Since JUnit 5 works
correctly with Maven, I don't think they will be motivated to dig for a
fix. I'll just have to try a different framework. It's too bad, I really
like the syntax. FYI, I tried both versions of the Surefire plugin and they
both failed.

**Comment by raniejade** [comment]
Just filed https://github.com/junit-team/junit5/issues/1130, hopefully the team can help shed some light on this.

**Comment by raniejade** [comment]
Fix is here: https://github.com/junit-team/junit5/pull/1131. Tested it locally and it works with @devonhumes demo project. Close this as it's not an issue with Spek.

**Comment by raniejade** [comment]
Should be fixed by the time JUnit Platform `1.1.0` is released.

**Comment by devonhumes** [comment]
Sounds good!
